vector/vector.go

```diff
@@ -0,0 +1,13 @@
+package vector
+
+import (
+	"github.com/brimdata/zed"
+)
+
+// len(values) == len(Types)
+type Vector struct {
+	Context *zed.Context
+	Types   []zed.Type
+	values  []any
```

Comment:
A comment immediately before an exported symbol is a [doc comment](https://tip.golang.org/doc/comment) and will appear in generated documentation. Sinc this one's about internals, hang it on `values` to keep it out of those docs.
```suggestion
type Vector struct {
	Context *zed.Context
	Types   []zed.Type
	// len(values) == len(types)
	values  []any
```